Speed up the gdal build by using all available cores for make.

lakinwecker opened the pull request:
I would sign the CLA - but I can't figure out how to do so. It seems to be busted in Firefox on linux.

I have reviewed the PDF and I would agree to it if I could.  I'm submitting the PR anyways, and maybe I can figure out how to sign the CLA later.

lakinwecker committed:
Speed up the gdal build by using all available cores for make.

lakinwecker commented:
Ha!  "All CLA requirements met".  I guess I signed it. >:)

lakinwecker committed:
Also build the debug build with all cores.

cbezault commented on ports/gdal/portfile.cmake:
```diff
@@ -332,6 +332,9 @@ elseif (VCPKG_CMAKE_SYSTEM_NAME STREQUAL "Linux" OR VCPKG_CMAKE_SYSTEM_NAME STRE
       message(FATAL_ERROR "MAKE not found")
   endif()
 
+  include(ProcessorCount)
```
While this is a cool new module we prefer the use of `VCPKG_CONCURRENCY` because it respects an environment variable that the user can customize.

cbezault commented on ports/gdal/portfile.cmake:
```diff
@@ -347,7 +350,7 @@ elseif (VCPKG_CMAKE_SYSTEM_NAME STREQUAL "Linux" OR VCPKG_CMAKE_SYSTEM_NAME STRE
 
     message(STATUS "Building ${TARGET_TRIPLET}-rel")
     vcpkg_execute_required_process(
```
We should also being using `vcpkg_execute_build_process` which adds some retry logic.

cbezault commented on ports/gdal/portfile.cmake:
```diff
@@ -347,7 +350,7 @@ elseif (VCPKG_CMAKE_SYSTEM_NAME STREQUAL "Linux" OR VCPKG_CMAKE_SYSTEM_NAME STRE
     message(STATUS "Building ${TARGET_TRIPLET}-rel")
     vcpkg_execute_required_process(
-      COMMAND make
+      COMMAND make "-j${NUM_CPUS}"
```
We should also provide a `NO_PARALLEL_COMMAND` that we will retry if there was a failure on the parallel command. We do this because of potential OOM issues when you run a bunch of linkers in parallel.

cbezault commented on ports/gdal/portfile.cmake:
```diff
@@ -383,7 +386,7 @@ elseif (VCPKG_CMAKE_SYSTEM_NAME STREQUAL "Linux" OR VCPKG_CMAKE_SYSTEM_NAME STRE
 
     message(STATUS "Building ${TARGET_TRIPLET}-dbg")
     vcpkg_execute_required_process(
```
Same as above.

lakinwecker committed:
Addresses the review comments.

Use VCPKG_CONCURRENCY so the normal vcpkg concurrency controls apply.
Use vcpkg_execute_build_process and NO_PARALLEL_COMMAND which allows
for builds to be retried in some cases where parallel builds might fail.

lakinwecker commented:
Thanks for the feedback, I think I've addressed all three points.
